tests/gtest/test_netif.cpp

```diff
@@ -297,4 +360,55 @@ TEST(Netif, WpanIfHasCorrectUnicastAddresses_AfterUpdatingUnicastAddresses)
 
     netif.Deinit();
 }
+
+TEST(Netif, WpanIfHasCorrectMulticastAddresses_AfterUpdatingMulticastAddresses)
+{
+    const char *wpan = "wpan0";
+    otbr::Netif netif;
+    EXPECT_EQ(netif.Init(wpan), OT_ERROR_NONE);
+
+    const char *kDefaultMulAddr1Str = "ff01::1";
+    const char *kDefaultMulAddr2Str = "ff02::1";
+    const char *kDefaultMulAddr3Str = "ff02::2";
+
+    otbr::Ip6Address kMulAddr1 = {
+        {0xff, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xfc}};
+    otbr::Ip6Address kMulAddr2 = {
+        {0xff, 0x32, 0x00, 0x40, 0xfd, 0x0d, 0x07, 0xfc, 0xa1, 0xb9, 0xf0, 0x50, 0x00, 0x00, 0x00, 0x01}};
+    const char *kMulAddr1Str = "ff03::fc";
+    const char *kMulAddr2Str = "ff32:40:fd0d:7fc:a1b9:f050:0:1";
+
+    otbr::Ip6Address testArray1[] = {
+        kMulAddr1,
+    };
+    std::vector<otbr::Ip6Address> testVec1(testArray1, testArray1 + sizeof(testArray1) / sizeof(otbr::Ip6Address));
+    netif.UpdateIp6MulticastAddresses(testVec1);
```

Comment:
may also test the case that netif.UpdateIp6MulticastAddresses contains one of the default subscribed multicast addresses?

Reply:
Sounds good! Added a case for that.